src/open_inwoner/openzaak/clients.py

```diff
@@ -1,28 +1,565 @@
+import base64
 import logging
-from typing import Optional
+from datetime import date
+from typing import List, Optional
+
+from django.conf import settings
+from django.core.files.uploadedfile import InMemoryUploadedFile
+from django.utils.functional import SimpleLazyObject
 
 from ape_pie.client import APIClient
+from requests import HTTPError, RequestException, Response
+from zgw_consumers.api_models.base import factory
+from zgw_consumers.api_models.catalogi import Catalogus
+from zgw_consumers.api_models.constants import RolOmschrijving
 from zgw_consumers.client import build_client as _build_client
+from zgw_consumers.service import pagination_helper
 
-from open_inwoner.utils.api import JSONParserClient
+from open_inwoner.openzaak.api_models import InformatieObject
+from open_inwoner.openzaak.models import OpenZaakConfig
+from open_inwoner.utils.api import ClientError, get_json_response
 
+from ..utils.decorators import cache as cache_result
+from .api_models import (
+    InformatieObjectType,
+    OpenSubmission,
+    Resultaat,
+    ResultaatType,
+    Rol,
+    Status,
+    StatusType,
+    Zaak,
+    ZaakInformatieObject,
+    ZaakType,
+)
 from .models import OpenZaakConfig
 
+CRS_HEADERS = {"Content-Crs": "EPSG:4326", "Accept-Crs": "EPSG:4326"}
+
 logger = logging.getLogger(__name__)
 
 
+class ZakenClient(APIClient):
+    @cache_result(
+        "cases:{user_bsn}:{max_requests}:{identificatie}",
+        timeout=settings.CACHE_ZGW_ZAKEN_TIMEOUT,
+    )
+    def fetch_cases(
```

Comment:
We might want to rename this to be a bit more standardized (because now I see things like case / zaak mixed). Perhaps something like create/read/update/delete + name of the resource in the API: `retrieve_zaken`